distributed/remote.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import os
+import subprocess
+import tempfile
+
+from tornado import gen
+
+from tornado.ioloop import IOLoop
+
+from distributed.compatibility import unicode
+from distributed.core import Server, write
+from distributed.utils import get_ip
+
+logger = logging.getLogger(__name__)
+
+class RemoteClient(Server):
+    def __init__(self, ip=get_ip(), local_dir=tempfile.mkdtemp(prefix='client-'), loop=None, **kwargs):
+        self.ip = ip
+        self.loop = loop or IOLoop.current()
+        self.local_dir = local_dir
+        handlers = {'upload_file': self.upload_file, 'execute': self.execute}
+        super(RemoteClient, self).__init__(handlers, io_loop=self.loop, **kwargs)
+
+    @gen.coroutine
+    def _start(self, port=0):
+        self.listen(port)
+
+    def start(self,port=0):
+        self.loop.add_callback(self._start, port)
+        logger.info("Remote Client is running at {0}".format(port))
+
+
+    @gen.coroutine
+    def execute(self, stream=None, filename=None):
+        script_path = os.path.join(self.local_dir, filename)
+        cmd = 'python {0}'.format(script_path)
```

Comment:
Might want to try `sys.executable` to get the same `python` from which this was run.